gpMgmt/bin/gpssh-exkeys

```
@@ -650,6 +525,19 @@ try:
 
     discovered_authorized_keys_file = os.path.join(tempDir, 'authorized_keys')
 
+    ######################
+    #  step 0
+    #
+    #    Ensure the local host can password-less ssh into each remote host
+    for remoteHost in GV.allHosts:
+        cmd = ['ssh', remoteHost.host(), '-o', 'BatchMode=yes', '-o', 'StrictHostKeyChecking=yes',  'true']
+        p = subprocess.Popen(cmd, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+        stdout, stderr = p.communicate()
+        if p.returncode:
+            print >> sys.stderr, '[ERROR]: Failed to ssh to %s. %s' % (remoteHost.host(), stderr)
+            print >> sys.stderr, '[ERROR]: Expected passwordless ssh to host %s' % remoteHost.host()
+            sys.exit(1)
```

You could still try to connect with each host and collect the entire set of hosts that you cannot connect to.  This would likely be more useful to a DBA.  And then only exit() if at least one host cannot be connected to.

---

We think it's better to fail early.

---

Sure, it's a judgment call.  It's faster to fail early, and if there are only a few hosts that don't work 1-N, that's better.  If there are 50 that don't work, the customer might want to see a list rather than run the command 50 times.  I'm not sure which is better but don't feel strongly either way.